### PR TITLE
Feature/aa 35 endpoint to show booking changes history to agent

### PR DESCRIPTION
**Pull request by AlexeyShok**
Added endpoint to show booking changes history to agent

### CODE REVIEW

File: Api/Services/Accommodations/Bookings/Management/BookingInfoService.cs [hunk 1]
```diff
@@ -152,6 +153,24 @@ public IQueryable<SlimAccommodationBookingInfo> GetAgentBookingsInfo(AgentContex
         }
 
         
+        public async Task<Result<List<BookingStatusHistoryEntry>>> GetBookingStatusHistory(int bookingId, AgentContext agent)
+        {
+            var booking = await _context.Bookings
+                .SingleOrDefaultAsync(b => b.Id == bookingId && b.AgencyId == agent.AgencyId);
+            if (booking is null)
+                return Result.Failure<List<BookingStatusHistoryEntry>>($"Booking with ID {bookingId} not found");
```
Comment (DFSko) [inline Api/Services/Accommodations/Bookings/Management/BookingInfoService.cs:161]: Maybe "Id"?

File: Api/Services/Accommodations/Bookings/Management/BookingInfoService.cs [hunk 2]
```diff
@@ -152,6 +153,24 @@ public IQueryable<SlimAccommodationBookingInfo> GetAgentBookingsInfo(AgentContex
...
+
+            var history = await _context.BookingStatusHistory
+                .Where(bsh => bsh.BookingId == bookingId)
+                .OrderBy(bsh => bsh.Id)
+                .ToListAsync();
+
+            return history is null
```
Comment (DFSko) [inline Api/Services/Accommodations/Bookings/Management/BookingInfoService.cs:168]: You can use null-coalescing here
`return history ?? new();`

File: Api/Services/Accommodations/Bookings/Management/BookingInfoService.cs (hunk 1 above)
Comment (AlexeyShok) [reply Api/Services/Accommodations/Bookings/Management/BookingInfoService.cs]: changed

File: Api/Services/Accommodations/Bookings/Management/BookingInfoService.cs (hunk 2 above)
Comment (AlexeyShok) [reply Api/Services/Accommodations/Bookings/Management/BookingInfoService.cs]: changed

**Review by DFSko — APPROVED** [review]

File: Api/Services/Accommodations/Bookings/Management/BookingInfoService.cs [hunk 3]
```diff
@@ -152,6 +153,22 @@ public IQueryable<SlimAccommodationBookingInfo> GetAgentBookingsInfo(AgentContex
         }
 
         
+        public async Task<Result<List<BookingStatusHistoryEntry>>> GetBookingStatusHistory(int bookingId, AgentContext agent)
```
Comment (mingazhev) [inline Api/Services/Accommodations/Bookings/Management/BookingInfoService.cs:156]: Booking could be passed there with no need to get another time

Comment (kirillta) [reply Api/Services/Accommodations/Bookings/Management/BookingInfoService.cs:156]: agree

Comment (AlexeyShok) [reply Api/Services/Accommodations/Bookings/Management/BookingInfoService.cs:156]: I removed the booking receipt from the service. No need to re-check, as CheckPermissions does this already

**Review by mingazhev — APPROVED** [review]

**Pull request merged**